From 01e3041b5716382f653c49538389230a8475f574 Mon Sep 17 00:00:00 2001
From: Robin Lee <robinlee.sysu@gmail.com>
Date: Sun, 26 Feb 2023 13:56:14 +0800
Subject: [PATCH 042/381] ec: Use .machine "any" explicitly in
 ecp_nistp521-ppc64

Since GCC commit e154242724b084380e3221df7c08fcdbd8460674 the flag "-many"
is sometimes not passed to the assembler. Use .machine "any" just like
ecp_nistz256-ppc64 to prevent compile errors when built with some
configurations of GCC.

CLA: trivial

Reviewed-by: Matt Caswell <matt@openssl.org>
Reviewed-by: Paul Dale <pauli@openssl.org>
(Merged from https://github.com/openssl/openssl/pull/20378)

(cherry picked from commit a6d7093a1dc890493d3577c738e729c0265d7b40)
---
 crypto/ec/asm/ecp_nistp521-ppc64.pl | 1 +
 1 file changed, 1 insertion(+)

diff --git a/crypto/ec/asm/ecp_nistp521-ppc64.pl b/crypto/ec/asm/ecp_nistp521-ppc64.pl
index 4260e24a1f..f8393e465d 100755
--- a/crypto/ec/asm/ecp_nistp521-ppc64.pl
+++ b/crypto/ec/asm/ecp_nistp521-ppc64.pl
@@ -140,6 +140,7 @@ ___
 }
 
 $code.=<<___;
+.machine	"any"
 .text
 
 ___
-- 
2.34.1

